### PR TITLE
.NET Standard SDK 4.1.0

### PR DESCRIPTION
**Pull request by mediumTaj**
### Summary
This pull request releases the .NET SDK version 4.1.0. This release includes support for Discovery v2 (CPD only), Assistant `alternate_intents` support and VisualRecognition v4 `training_usage` support.

### CODE REVIEW

File: src/IBM.Watson.VisualRecognition.v4/VisualRecognitionService.cs [hunk 1]
```diff
@@ -105,16 +105,22 @@ public DetailedResponse<AnalyzeResponse> Analyze(List<string> collectionIds, Lis
 
                 if (collectionIds != null)
                 {
-                    var collectionIdsContent = new StringContent(string.Join(",", collectionIds.ToArray()), Encoding.UTF8, HttpMediaType.TEXT_PLAIN);
-                    collectionIdsContent.Headers.ContentType = null;
-                    formData.Add(collectionIdsContent, "collection_ids");
+                    foreach (string item in collectionIds)
```
Comment (lpatino10) [inline src/IBM.Watson.VisualRecognition.v4/VisualRecognitionService.cs]: Do we not need to make this handwritten change anymore for the form params?

File: test/Discovery.v1.IntegrationTests/DiscoveryIntegrationTests.cs [hunk 2]
```diff
@@ -406,6 +406,51 @@ public void TestQuery()
         }
         #endregion
 
+        #region Query Aggregation
```
Comment (lpatino10) [inline test/Discovery.v1.IntegrationTests/DiscoveryIntegrationTests.cs]: Good that you have this in here now 👍

File: test/ToneAnalyzer.v3.UnitTests/ToneAnalyzerServiceUnitTests.cs [hunk 3]
```diff
@@ -1,410 +0,0 @@
...
-using Microsoft.VisualStudio.TestTools.UnitTesting;
-using NSubstitute;
-using System;
-using System.Collections.Generic;
-using System.Net;
-using System.Net.Http;
-using System.Threading.Tasks;
-
-
-namespace IBM.Watson.ToneAnalyzer.v3.UnitTests
```
Comment (lpatino10) [inline test/ToneAnalyzer.v3.UnitTests/ToneAnalyzerServiceUnitTests.cs]: Did you mean to delete this? I didn't see any of the old handwritten unit test files get deleted.

File: examples/IBM.Watson.Discovery.v2.Examples/DiscoveryTestData/exampleConfigurationData.json [hunk 4]
```diff
@@ -0,0 +1,133 @@
+{
```
Comment (lpatino10) [inline examples/IBM.Watson.Discovery.v2.Examples/DiscoveryTestData/exampleConfigurationData.json]: This and the other Discovery v2 example data files don't seem to get used anywhere.

File: src/IBM.Watson.VisualRecognition.v4/VisualRecognitionService.cs (hunk 1 above)
Comment (mediumTaj) [reply src/IBM.Watson.VisualRecognition.v4/VisualRecognitionService.cs]: Not in V4 - The service accepts both comma separated string and multiple form parts for collectionIds and features. We do still need a handwritten change for V3 though. It is being generated as multiple form parts but we need it to be sent as comma separated string. There's been activity in OpenApi to support these form params as comma separated strings in the next OpenApi release.

File: test/ToneAnalyzer.v3.UnitTests/ToneAnalyzerServiceUnitTests.cs (hunk 3 above)
Comment (mediumTaj) [reply test/ToneAnalyzer.v3.UnitTests/ToneAnalyzerServiceUnitTests.cs]: Yeah, this one was problematic and commented out anyway (line 33).

File: examples/IBM.Watson.Discovery.v2.Examples/DiscoveryTestData/exampleConfigurationData.json (hunk 4 above)
Comment (mediumTaj) [reply examples/IBM.Watson.Discovery.v2.Examples/DiscoveryTestData/exampleConfigurationData.json]: Ah you're right - I must've copied the whole directory. I will remove.

File: src/IBM.Watson.VisualRecognition.v4/VisualRecognitionService.cs (hunk 1 above)
Comment (lpatino10) [reply src/IBM.Watson.VisualRecognition.v4/VisualRecognitionService.cs]: Ah, good to know! Should be good then 👍

File: src/IBM.Watson.Common/Common.cs [hunk 5]
```diff
@@ -27,7 +27,7 @@ public partial class Common
         /// <summary>
         /// The SDK version.
         /// </summary>
-        public const string Version = "watson-apis-dotnet-standard-sdk-4.0.0";
```
Comment (lpatino10) [inline src/IBM.Watson.Common/Common.cs]: It seems like this and the version in the `Common.csproj` could also be taken care of with bumpversion, but not something that needs to be done now.

**Comment by lpatino10** [comment]
@mediumTaj did you take a look at what Phil did in the Python core in regards to releases? https://github.com/IBM/python-sdk-core/blob/master/.releaserc

It looks like it solved the problem of bumpversion updates not propagating back to `master` after the release. If you haven't made that change it might be worth doing for this release.

File: src/IBM.Watson.Common/Common.cs (hunk 5 above)
Comment (mediumTaj) [reply src/IBM.Watson.Common/Common.cs]: Yeah unfortunately the common version cannot be included in bumpversion. Common and core needs to be available on Nuget before CI will pass so I deployed them both already. I probably should break common out into its own repo eventually

**Comment by mediumTaj** [comment]
@lpatino10 I took a look at it but I never had any issues with the .NET SDK. It always merged into master fine - I keep checking just in case. I'll look to make this revision if I start having issues.

**Pull request merged**